### PR TITLE
Switch to GitHub changelog generator

### PR DESCRIPTION
**Pull request by bentranter**
Switches from github-release-notes to our newly open-sourced github-changelog-generator. Follows the pattern I used with mockgen, where we build the binary from the vendored library, and then delete it after the command has been run so we don't pollute the working directory or overwrite anything in the user's `$GOBIN` directory.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,7 @@ require (
 	github.com/BurntSushi/toml v0.3.1 // indirect
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/creack/pty v1.1.7
+	github.com/digitalocean/github-changelog-generator v0.0.0-20190822155117-98248440dda1
```
Comment (hilary) [inline go.mod]: I don't think you want ghcg in `go.mod` :)

File: scripts/changelog.sh [hunk 2]
```diff
@@ -3,8 +3,6 @@
 set -o pipefail
 
 tfile=$(mktemp /tmp/doctl-CHANGELOG-XXXXXX)
-github-release-notes -org digitalocean -repo doctl -since-latest-release -include-author >"$tfile"
-
-GO111MODULE=on go mod tidy
```
Comment (hilary) [inline scripts/changelog.sh]: You'll need the `go mod tidy` to prevent ghcg from ending up in `go.mod`

File: vendor/modules.txt [hunk 3]
```diff
@@ -4,6 +4,9 @@ github.com/blang/semver
 github.com/creack/pty
 # github.com/davecgh/go-spew v1.1.1
 github.com/davecgh/go-spew/spew
+# github.com/digitalocean/github-changelog-generator v0.0.0-20190822155117-98248440dda1
+github.com/digitalocean/github-changelog-generator
```
Comment (hilary) [inline vendor/modules.txt]: These are here because they are in `go.mod`

File: scripts/changelog.sh (hunk 2 above)
Comment (bentranter) [reply scripts/changelog.sh]: This is done, not sure why GitHub isn't picking up the change

Comment (hilary) [reply scripts/changelog.sh]: I see it :)

**Pull request merged**